From 35acb423f360f69645f6ba50552090d04a6280ad Mon Sep 17 00:00:00 2001
From: Mateusz Polrola <mateuszx.potrola@intel.com>
Date: Fri, 22 Jun 2018 15:23:50 +0200
Subject: [PATCH 1565/2313] intel-ipu4: Fixed compilation error without
 DEBUG_FS

ipu_buttress_tsc_ticks_to_ns is currently guarded with ifdef for
CONFIG_DEBUG_FS, but it has not dependencies to it and is used also by
other parts of IPU code (that do not depend on DEBUG_FS),
causing compilation errors with DEBUG_FS disabled.

Change-Id: I9509e06e78deede61e1ed78a751097da296398c6
Signed-off-by: Mateusz Polrola <mateuszx.potrola@intel.com>
---
 drivers/media/pci/intel/ipu-buttress.c | 25 +++++++++++++------------
 1 file changed, 13 insertions(+), 12 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-buttress.c b/drivers/media/pci/intel/ipu-buttress.c
index 0f128f3..19c64f1 100644
--- a/drivers/media/pci/intel/ipu-buttress.c
+++ b/drivers/media/pci/intel/ipu-buttress.c
@@ -1432,6 +1432,19 @@ int ipu_buttress_tsc_read(struct ipu_device *isp, u64 *val)
 }
 EXPORT_SYMBOL_GPL(ipu_buttress_tsc_read);
 
+u64 ipu_buttress_tsc_ticks_to_ns(u64 ticks)
+{
+	/*
+	 * TSC clock frequency is 19.2MHz,
+	 * converting TSC tick count to ns is calculated by:
+	 * ns = ticks * 1000 000 000 / 19.2Mhz
+	 *    = ticks * 1000 000 000 / 19200000Hz
+	 *    = ticks * 10000 / 192 ns
+	 */
+	return ticks * 10000 / 192;
+}
+EXPORT_SYMBOL_GPL(ipu_buttress_tsc_ticks_to_ns);
+
 #ifdef CONFIG_DEBUG_FS
 
 static int ipu_buttress_reg_open(struct inode *inode, struct file *file)
@@ -1497,18 +1510,6 @@ static int ipu_buttress_start_tsc_sync_set(void *data, u64 val)
 DEFINE_SIMPLE_ATTRIBUTE(ipu_buttress_start_tsc_sync_fops, NULL,
 			ipu_buttress_start_tsc_sync_set, "%llu\n");
 
-u64 ipu_buttress_tsc_ticks_to_ns(u64 ticks)
-{
-	/*
-	 * TSC clock frequency is 19.2MHz,
-	 * converting TSC tick count to ns is calculated by:
-	 * ns = ticks * 1000 000 000 / 19.2Mhz
-	 *    = ticks * 1000 000 000 / 19200000Hz
-	 *    = ticks * 10000 / 192 ns
-	 */
-	return ticks * 10000 / 192;
-}
-EXPORT_SYMBOL_GPL(ipu_buttress_tsc_ticks_to_ns);
 
 static int ipu_buttress_tsc_get(void *data, u64 *val)
 {
-- 
2.7.4

